daisy_workflows/export/export_disk_ext.sh

```diff
@@ -17,28 +17,51 @@ set -x
 URL="http://metadata/computeMetadata/v1/instance/attributes"
 GS_PATH=$(curl -f -H Metadata-Flavor:Google ${URL}/gcs-path)
 FORMAT=$(curl -f -H Metadata-Flavor:Google ${URL}/format)
+DISK_RESIZING_MON=$(curl -f -H Metadata-Flavor:Google ${URL}/resizing-script-name)
 
 # Strip gs://
-LOCAL_PATH=${GS_PATH##*//}
+IMAGE_OUTPUT_PATH=${GS_PATH##*//}
 # Create dir for output
-OUTS_PATH=${LOCAL_PATH%/*}
+OUTS_PATH=${IMAGE_OUTPUT_PATH%/*}
 mkdir -p "/gs/${OUTS_PATH}"
 
-# Disk image size info.
-SIZE_BYTES=$(qemu-img info --output "json" /dev/sdb | grep -m1 "virtual-size" | grep -o '[0-9]\+')
- # Round up to the next GB.
-SIZE_GB=$(awk "BEGIN {print int((${SIZE_BYTES}/1000000000)+ 1)}")
+# Prepare disk size info.
+# 1. Disk image size info.
+SIZE_BYTES=$(lsblk /dev/sdb --output=size -b | sed -n 2p)
+# 2. Round up to the next GB.
+SIZE_OUTPUT_GB=$(awk "BEGIN {print int(((${SIZE_BYTES}-1)/1073741824) + 1)}")
+# 3. Add 5GB of additional space to max size to prevent the corner case that output
+# file is slightly larger than source disk.
+MAX_BUFFER_DISK_SIZE_GB=$(awk "BEGIN {print int(${SIZE_OUTPUT_GB} + 5)}")
 
-echo "GCEExport: Exporting disk of size ${SIZE_GB}GB and format ${FORMAT}."
+# Prepare buffer disk
+echo "GCEExport: Initializing buffer disk for qemu-img output..."
+sudo mkfs.ext4 /dev/sdc
+sudo mount /dev/sdc "/gs/${OUTS_PATH}"
 
-if ! out=$(qemu-img convert /dev/sdb "/gs/${LOCAL_PATH}" -p -O $FORMAT 2>&1); then
+# Fetch disk size monitor script from GCS1
+DISK_RESIZING_MON_GCS_PATH=gs://${OUTS_PATH%/*}/sources/${DISK_RESIZING_MON}
+DISK_RESIZING_MON_LOCAL_PATH=/gs/${DISK_RESIZING_MON}
+echo "GCEExport: Copying disk size monitor script..."
+if ! out=$(gsutil cp "${DISK_RESIZING_MON_GCS_PATH}" "${DISK_RESIZING_MON_LOCAL_PATH}" 2>&1); then
+  echo "ExportFailed: Failed to copy disk size monitor script. Error: ${out}"
+  exit
+fi
+echo ${out}
+
+echo "GCEExport: Launching disk size monitor in background..."
+bash ${DISK_RESIZING_MON_LOCAL_PATH} ${MAX_BUFFER_DISK_SIZE_GB} &
```

Comment:
np no need for bash prefix

Reply:
why did you resolve this?

Reply:
discussed irl, recommend replacing with `chmod +x ${DISK_RESIZING_MON_LOCAL_PATH}` after download

Reply:
modified as discussed